SYCL/ESIMD/api/functional/ctors/operator_assignment.hpp

```diff
@@ -0,0 +1,97 @@
+//===-- operator_assignment.hpp - Functions for tests on simd assignment
+//      operators. --------------------------------------------------------===//
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//===----------------------------------------------------------------------===//
+///
+/// \file
+/// This file provides functios for tests on simd assignment operators.
+///
+//===----------------------------------------------------------------------===//
+
+#pragma once
+
+// The test proxy is used to verify the move constructor was called actually.
+#define __ESIMD_ENABLE_TEST_PROXY
+
+#include "../shared_element.hpp"
+#include "common.hpp"
+
+namespace esimd_test::api::functional::ctors {
+
+// The main test routine.
+// Using functor class to be able to iterate over the pre-defined data types.
+template <typename DataT, int NumElems, typename TestCaseT,
+          typename ShouldMoveCtorBeCalled>
```

Comment:
```suggestion
template <typename DataT, int NumElems, typename TestCaseT>
```
Isn't a requirement to call or not to call `move assignment operator` a part of the test case, as result - a part of `testCaseT` interface?
Please, verify you don't mention constructors instead of operators anywhere.